### PR TITLE
Launch storage for ocis-metadata

### PR DESCRIPTION
**Pull request by IljaN**

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

File: pkg/command/storagemetadata.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+						},
+					},
+					"http": map[string]interface{}{
+						"network": cfg.Reva.StorageMetadataData.Network,
+						"address": cfg.Reva.StorageMetadataData.Addr,
+						// TODO build services dynamically
+						"services": map[string]interface{}{
+							"dataprovider": map[string]interface{}{
+								"prefix":      "data",
+								"driver":      "local",
```
Comment (IljaN) [inline pkg/command/storagemetadata.go]: Ah config value missing here

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/flagset/storagemetadata.go  1
- pkg/command/storagemetadata.go  8
         

Clones added
============
- pkg/flagset/storagemetadata.go  2
- pkg/command/storagemetadata.go  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

File: pkg/command/storagemetadata.go (hunk 1 above)
Comment (IljaN) [reply pkg/command/storagemetadata.go]: Fixed

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

File: pkg/command/storagehomedata.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/owncloud/ocis-reva/pkg/server/debug"
 )
 
-// StorageHomeData is the entrypoint for the storage-home-data command.
+// StorageHomeData is the entrypoint for the storage-home data command.
```
Comment (butonic) [inline pkg/command/storagehomedata.go]: the command really is called `storage-home-data` ...

File: pkg/command/storagemetadata.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	"github.com/cs3org/reva/cmd/revad/runtime"
+	"github.com/gofrs/uuid"
+	"github.com/micro/cli/v2"
+	"github.com/oklog/run"
+	"github.com/owncloud/ocis-reva/pkg/config"
+	"github.com/owncloud/ocis-reva/pkg/flagset"
+	"github.com/owncloud/ocis-reva/pkg/server/debug"
+)
+
+// StorageMetadata the entrypoint for the reva-storage-meta-data command.
```
Comment (butonic) [inline pkg/command/storagemetadata.go]: `Name:     "reva-storage-metadata",` no `-` between meta and data

File: pkg/command/storagemetadata.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+				//metrics     = metrics.New()
+			)
+
+			defer cancel()
+
+			{
+				uuid := uuid.Must(uuid.NewV4())
+				pidFile := path.Join(os.TempDir(), "revad-"+c.Command.Name+"-"+uuid.String()+".pid")
+
+				cfg.Reva.Storages.Common.EnableHome = false
```
Comment (butonic) [inline pkg/command/storagemetadata.go:74]: add comment why the metadata storage does not need 'enable home'

File: pkg/config/config.go [hunk 5]
```diff
@@ -101,6 +101,14 @@ type PublicStorage struct {
 	UserProviderAddr        string
 }
 
+// MetadataStorage configures a metadata-storage provider
+type MetadataStorage struct {
```
Comment (butonic) [inline pkg/config/config.go]: AFAICT unused. and you really just need a StoragePort. get rid of this ;-)

File: pkg/flagset/storagemetadata.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+			Value:       "local",
+			Usage:       "storage driver for data-provider mount: eg. local, eos, owncloud, ocis or s3",
+			EnvVars:     []string{"REVA_STORAGE_METADATA_DATA_PROVIDER_DRIVER"},
+			Destination: &cfg.Reva.StorageMetadata.Driver,
+		},
+	}
+
+	flags = append(flags, TracingWithConfig(cfg)...)
+	flags = append(flags, DebugWithConfig(cfg)...)
+	flags = append(flags, SecretWithConfig(cfg)...)
```
Comment (butonic) [inline pkg/flagset/storagemetadata.go:64]: missing for driver configuration:
```

	flags = append(flags, DriverEOSWithConfig(cfg)...)
	flags = append(flags, DriverLocalWithConfig(cfg)...)
	flags = append(flags, DriverOwnCloudWithConfig(cfg)...)
	flags = append(flags, DriverOCISWithConfig(cfg)...)
```

Comment (butonic) [reply pkg/flagset/storagemetadata.go:64]: you may want to overwrite the default configs with different options for the root, eg
`REVA_STORAGE_OCIS_ROOT` should not use the same dir as the normal ocis driver.

same for the local, eos and owncloud drivers.

**Review by butonic — CHANGES_REQUESTED** [review]

**Comment by butonic** [comment]
other than the above :rocket:

File: pkg/flagset/storagemetadata.go (hunk 6 above)
Comment (IljaN) [reply pkg/flagset/storagemetadata.go:64]: :heavy_check_mark:

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  1
         

Clones added
============
- pkg/command/storagemetadata.go  8
- pkg/flagset/storagemetadata.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=6189978&bid=20197448)

**Comment by IljaN** [comment]
Continued here: https://github.com/owncloud/ocis/pull/602

**Pull request closed without merging**